src/chainstate/stacks/index/bits.rs

```diff
@@ -232,86 +197,54 @@ pub fn ptrs_from_bytes<R: Read>(node_id: u8, r: &mut R, ptrs_buf: &mut [TriePtr]
 }
 
 /// Calculate the hash of a TrieNode, given its childrens' hashes.
-pub fn get_node_hash<T: TrieNode + std::fmt::Debug>(node: &T, child_hashes: &Vec<TrieHash>) -> TrieHash {
+pub fn get_node_hash<M, T: ConsensusSerializable<M> + std::fmt::Debug>(node: &T, child_hashes: &Vec<TrieHash>, map: &M) -> TrieHash {
```

Comment:
Is there a reason the `M` type parameter needs to exist here?

Reply:
Yeah, it’s so that the same node hashing routines can be used for Merkle proof tries (which embed the BlockHeaderHashes directly and therefore implement ConsensusSerializable with a () type for M, whereas the TrieNode type requires the BlockHashMap)

Reply:
Ah, gotcha.  Makes sense :+1: Maybe add a comment explaining this if you have a second?

Reply:
Comments added in d31f0207